src/main/java/org/broadinstitute/hellbender/engine/TwoPassReadWalker.java

```diff
@@ -34,6 +34,11 @@ public void traverse() {
         traverseReads(countedFilter, this::firstPassApply);
         logger.info("Finished First Pass");
         afterFirstPass();
+        // Need to reinitilize the reads and intervals so they are guarinteed to pass over a file
+        initializeReads();
+        if ( hasIntervals() ) {
```

Comment:
could you extract this as a package private function `setReadTraversalBounds()` in ReadWalker since it's just a copy from their?